ironfish-rust/src/frost_utils/signing_package.rs

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::{errors::IronfishError, transaction::unsigned::UnsignedTransaction};
+use ironfish_frost::frost::SigningPackage as FrostSigningPackage;
+
+#[derive(Clone)]
+pub struct SigningPackage {
+    pub unsigned_transaction: UnsignedTransaction,
+    pub frost_signing_package: FrostSigningPackage,
+}
+
+impl SigningPackage {
+    pub fn serialize(&self) -> Result<Vec<u8>, IronfishError> {
+        let serialized_frost_signing_package = self.frost_signing_package.serialize()?;
+        let length_bytes = (serialized_frost_signing_package.len() as u32).to_le_bytes();
+
+        let mut vec: Vec<u8> = vec![];
+        // write length
+        vec.extend_from_slice(&length_bytes);
+        // write frost signing package
+        vec.extend_from_slice(&serialized_frost_signing_package);
+
+        let mut transaction_vec = vec![];
+        // write unsigned transaction
+        self.unsigned_transaction.write(&mut transaction_vec)?;
+
+        vec.extend_from_slice(&transaction_vec);
+
+        Ok(vec)
+    }
+
+    pub fn deserialize(bytes: &[u8]) -> Result<SigningPackage, IronfishError> {
+        let mut signing_package_length_bytes = [0u8; 4];
+        signing_package_length_bytes.copy_from_slice(&bytes[..4]);
+
+        let length = u32::from_le_bytes(signing_package_length_bytes);
+
+        let frost_signing_package_bytes = &bytes[4..(4 + length as usize)];
```

Comment:
the byte offsets in here look a bit odd

i wonder if it might be cleaner to follow the reader/writer pattern that we've used in other serializations like `SigningCommitment`

Reply:
This is more consistent with Signature Share. I don't think doing read/ write will help with the "oddness". I think the core issue is that we don't know how large the unsigned transaction is going to be

Reply:
I'm not opposed to changing it to the read/ write pattern.

Reply:
> I don't think doing read/ write will help with the "oddness".

I think using a reader will let us rely on a cursor for reading through the byte array rather than slicing the array with indices.  It might make it a bit easier to follow since that's the pattern that most of our encodings use

Reply:
I'll give that a go